src/minigames/keyboardMinigame.ts

```diff
@@ -3,29 +3,33 @@ import { Minigame } from "./minigame";
 
 export class KeyboardMinigame extends Minigame {
   private keyDownListener = (e: KeyboardEvent) => void this.onKeyDown(e.key);
-  private currentLetter = (Math.floor(Math.random() * 10)).toString();
 
-  protected onKeyDown(key: string) {
-    console.log(key, 'pressed');
-    this.finishMinigame(key === this.currentLetter);
+  override attach() {
+    super.attach();
+    window.addEventListener('keydown', this.keyDownListener);
+  }
+
+  override detach() {
+    window.removeEventListener('keydown', this.keyDownListener);
+    super.detach();
   }
 
-  override populateContainer(): void {
+  protected override populateContainer(): void {
     const text = new Text({
-      text: this.currentLetter,
+      label: 'text',
+      text: (Math.floor(Math.random() * 10)).toString(),
       style: new TextStyle({ fill: '#de3249' })
     });
 
     this.container.addChild(text);
   }
 
-  override attach() {
-    super.attach();
-    window.addEventListener('keydown', this.keyDownListener);
-  }
-
-  override detach() {
-    window.removeEventListener('keydown', this.keyDownListener);
-    super.detach();
+  protected onKeyDown(key: string) {
+    console.log(key, 'pressed');
+    const text = this.container.getChildByLabel('text');
+    if (!text || !(text instanceof Text)) {
+      this.finishMinigame(false);
```

Comment:
Lol if we have a internal error the player loses. I like it,

Reply:
It's their fault!